### PR TITLE
Display json report regardless of convert2rhel output

### PR DESCRIPTION
**Pull request by r0x0d**
[HMS-3046](https://issues.redhat.com/browse/HMS-3046)
After we are done with pre-conversion or conversion, we will output the generated report json if that exists.

### CODE REVIEW

**Comment by mshriver** [comment]
https://issues.redhat.com/browse/HMS-3046

File: scripts/conversion_script.py [hunk 1]
```diff
@@ -428,6 +442,12 @@ def update_insights_inventory():
 
 def main():
     """Main entrypoint for the script."""
+    if os.path.exists(C2R_REPORT_FILE):
+        archive_analysis_report(C2R_REPORT_FILE)
+
+    if os.path.exists(C2R_REPORT_TXT_FILE):
+        archive_analysis_report(C2R_REPORT_TXT_FILE)
```
Comment (andywaltlova) [inline scripts/conversion_script.py:449]: I am just thinking what happens when some exception occurs in these calls, lets say report exists but somehow the archive function fails, the move not successful or the archive folder can't be created, would it be worth it to wrap it in try block? and do something based on expectations, do we expect that if this fails the script should fail and report the issue?

Maybe it's not necessary, if it's run under root it should work

File: scripts/conversion_script.py [hunk 2]
```diff
@@ -482,6 +487,31 @@ def main():
...
+
+            if not output.report:
+                # Try to attach the textual report in the report if we have
+                # json report, otherwise, we would overwrite the report raised
+                # by the exception.
+                output.report = gather_textual_report()
+
+            output.entries = transform_raw_data(data)
+
+        update_insights_inventory()
```
Comment (andywaltlova) [inline scripts/conversion_script.py:513]: This being here means that we will try to update insights even when the fail doesn't generate new report - maybe the preconversion/conversion didn't even run, is that okay? It shouldn't break anything :shrug: just additional call that may not bring new information.

**Review by andywaltlova — APPROVED** [review]
Just two questions regarding the edge cases, but this version should work for the regular script use.

**Comment by r0x0d** [comment]
Closed this in favor of #36

**Pull request closed without merging**